salt_resources/states/basebox/miniconda.sls

```diff
@@ -0,0 +1,9 @@
+download_miniconda_installer:
+  cmd.run:
+    - name: wget http://repo.continuum.io/miniconda/Miniconda-latest-Linux-x86_64.sh -O /home/vagrant/miniconda.sh
+    - unless: test -f /usr/local/bin/miniconda/bin/conda
+
+ensure_miniconda_is_installed:
+  cmd.run:
+    - name: bash /home/vagrant/miniconda.sh -b -p /usr/local/bin/miniconda
```

Comment:
"source activate miniconda_env" does not work because the activate command is not registered in vagrant's bash profile. I do not know hoe to reload vagrant's bash shell.

You can run "/usr/local/bin/miniconda/bin/activate miniconda_env"  or "/usr/local/bin/miniconda/bin/conda"